toolbox/fdc3-for-web/fdc3-web-impl/src/ServerContext.ts

```diff
@@ -36,21 +37,18 @@ export interface ServerContext<X extends AppRegistration> {
    */
   post(message: object, instanceId: InstanceID): Promise<void>;
 
-  /**
-   * Post an outgoing message to a particular app
-   */
-  post(message: object, instanceId: InstanceID): Promise<void>;
-
   /**
    * Opens a new instance of an application.
    * Promise completes once the application window is opened
    */
   open(appId: string): Promise<InstanceID>;
 
-  /**
-   * Handle clean-up of state after an app instance disconnects.
+  /** Set the FDC3Server instance associated with this context. This reference is
+   *  used to notify the server to cleanup state for apps that have been terminated.
+   *  The FDC3Server is passed a ServerContext when created and should call this fn
+   *  in its constructor.
    */
-  goodbye(instanceId: string): void;
+  setFDC3Server(server: FDC3Server): void;
```

Comment:
rather than introduce the FDC3Server here, it might be better to have cleanup involve just the ServerContext and MessageHandler classes.   setAppState() on the ServerContext can then chain cleanup on the MessageHandlers if it wants to.

Reply:
yes, I agree about the problem.  This is as a result of cleanup needing to work across all the handlers.  The `ServerContext` is separated out from the handler as it's expected that an implementation of the Desktop agent Server (in this case, `demo`), will want to implement that in it's own way.  Whereas the behaviour of opening / resolving intents etc.  will be reused.

> If you don't want the ServerContext linked back to the server, then it will need to be linked to all the handlers so it can notify them to clean up their state (which is all that link is for) - that may be worse...

I think it would be less bad to have two classes referencing each other than three.  But if you don't think this would be an improvement then just leave it.